shim/src/main.rs

```diff
@@ -68,38 +84,111 @@ fn run_current_turbo(args: Vec<String>) -> Result<i32> {
     Ok(exit_code)
 }
 
-/// Checks if we are in single package mode by first seeing if there is a turbo.json
-/// in the ancestor path, and then checking for workspaces.
+/// Finds local turbo path given the package.json path. We assume that the node_modules directory
+/// is at the same level as the package.json file.
 ///
 /// # Arguments
 ///
-/// * `current_dir`: Current working directory
+/// * `package_json_path`: The location of the package.json file
 ///
-/// returns: Result<bool, Error>
+/// returns: Result<Option<PathBuf>, Error>
 ///
-fn is_single_package_mode(current_dir: &Path) -> Result<bool> {
-    let has_turbo_json = AncestorSearch::new(current_dir.to_path_buf(), "turbo.json")?
-        .next()
-        .is_some();
+fn find_local_turbo_path(repo_root: &Path) -> Result<Option<PathBuf>> {
+    let package_json_path = repo_root.join("package.json");
+    let package_json_contents = fs::read_to_string(&package_json_path)?;
+    let package_json: PackageJson = serde_json::from_str(&package_json_contents)?;
+
+    let dev_dependencies_has_turbo = package_json
+        .dev_dependencies
+        .map_or(false, |deps| deps.contains_key("turbo"));
+    let dependencies_has_turbo = package_json
+        .dependencies
+        .map_or(false, |deps| deps.contains_key("turbo"));
 
-    if has_turbo_json {
-        return Ok(false);
+    if dev_dependencies_has_turbo || dependencies_has_turbo {
+        let mut local_turbo_path = repo_root.join("node_modules");
+        local_turbo_path.push(".bin");
+        local_turbo_path.push("turbo");
+
+        Ok(Some(local_turbo_path))
+    } else {
+        Ok(None)
     }
+}
 
-    // We should detect which package manager and then determine workspaces from there,
-    // but detection is not implemented yet and really we're either checking the `package.json`
-    // or the `pnpm-workspace.yaml` file so we can do both.
-    let npm = PackageManager::Npm;
-    if npm.get_workspace_globs(current_dir).is_ok() {
-        return Ok(false);
-    };
+impl RepoState {
```

Comment:
I originally had `RepoType` instead of `RepoMode`. For this struct I'd say `RepoState` is better just cause the root path doesn't fall under a type, as every repo has a root. Although `RepoState` isn't fantastic because it implies some kind of state. I'll try to think of a better name